lib/src/insertion_order_parser.dart

```diff
@@ -0,0 +1,164 @@
+  }
+
+  /// Creates an [InsertionOrder_Pacing] instance from [map].
+  static InsertionOrder_Pacing _createPacing(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) return InsertionOrder_Pacing();
+
+    final pacing = InsertionOrder_Pacing()
+      ..pacingPeriod = _createPacingPeriod(map['pacingPeriod'])
+      ..pacingType = _createPacingType(map['pacingType'])
+      ..dailyMaxMicros = map['dailyMaxMicros'] ?? _emptyEntry
+      ..dailyMaxImpressions = map['dailyMaxImpressions'] ?? _emptyEntry;
+
+    return pacing;
+  }
+
+  /// Creates an [InsertionOrder_InsertionOrderBudget] instance from [map].
+  static InsertionOrder_InsertionOrderBudget _createBudget(
+      Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      return InsertionOrder_InsertionOrderBudget();
+    }
+
+    final budget = InsertionOrder_InsertionOrderBudget()
+      ..budgetUnit = _createBudgetUnit(map['budgetUnit'])
+      ..automationType = _createAutomationType(map['automationType']);
+
+    for (Map segmentMap in map['budgetSegments'] ?? []) {
+      budget.budgetSegments.add(_createBudgetSegment(segmentMap));
+    }
+    return budget;
+  }
+
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment
+      _createBudgetSegment(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      return InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment();
+    }
+
+    final budgetSegment =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment()
+          ..budgetAmountMicros = map['budgetAmountMicros'] ?? _emptyEntry
+          ..description = map['description'] ?? _emptyEntry
+          ..campaignBudgetId = map['campaignBudgetId'] ?? _emptyEntry
+          ..dateRange = _createDateRange(map['dateRange']);
+
+    return budgetSegment;
+  }
+
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange
+      _createDateRange(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      return InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange();
+    }
+
+    final dateRange =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange()
+          ..startDate = _createDate(map['startDate'])
+          ..endDate = _createDate(map['endDate']);
+
+    return dateRange;
+  }
+
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date
+      _createDate(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      return InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date();
+    }
+
+    final date =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date()
+          ..year = map['year'] ?? _emptyEntry
+          ..month = map['month'] ?? _emptyEntry
+          ..day = map['day'] ?? _emptyEntry;
+
+    return date;
+  }
+
+  static InsertionOrder_EntityStatus _createEntityStatus(String target) {
```

Comment:
consider: rather than having the one line method just doing it inline. not required though.

also: is there a default value method on proto enums?

Reply:
I don't want to believe this either but I didn't find a default value method :(

Instead of enum.values[0], would enum.valueOf(0) sounds more like a default value?

Reply:
By convention the '0' value is usally the default, however I think it's better to be explicit here as not all of these enums are doing this, so something like this instead:
   return ENTITY_STATUS_MAP[target] ?? InsertionOrder_EntityStatus.UNKOWN